controllers/serviceaccount/serviceaccount_creator.go

```diff
@@ -269,42 +183,44 @@ func (c *creator) executeCommand(ctx context.Context, consumerDogu *core.Dogu, s
 	return saCreds, nil
 }
 
-func (c *creator) getPublicKey(doguConfig registry.ConfigurationContext) (*keys.PublicKey, error) {
-	keyProviderStr, err := c.registry.GlobalConfig().Get("key_provider")
-	if err != nil {
-		return nil, fmt.Errorf("failed to get key provider: %w", err)
-	}
-	keyProvider, err := keys.NewKeyProvider(keyProviderStr)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create keyprovider: %w", err)
-	}
-	pubKeyStr, err := doguConfig.Get(registry.KeyDoguPublicKey)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get dogu public key: %w", err)
+func (c *creator) writeServiceAccounts(ctx context.Context, senDoguCfg *config.DoguConfig, serviceAccount core.ServiceAccount, saCreds map[string]string) error {
+	for key, value := range saCreds {
+		path := "/sa-" + serviceAccount.Type + "/" + key
+
+		updatedCfg, err := senDoguCfg.Set(config.Key(path), config.Value(value))
+		if err != nil {
+			return fmt.Errorf("failed to set sa value of key %s to sensisitive dogu config: %w", key, err)
+		}
+
+		senDoguCfg.Config = updatedCfg
 	}
-	publicKey, err := keyProvider.ReadPublicKeyFromString(pubKeyStr)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read public key from string: %w", err)
+
+	if err := writeConfig(ctx, senDoguCfg, c.sensitiveDoguRepo); err != nil {
+		return fmt.Errorf("failed write config for dogu %s after updating: %w", senDoguCfg.DoguName, err)
 	}
 
-	return publicKey, nil
+	return nil
 }
 
-func (c *creator) writeServiceAccounts(doguConfig registry.ConfigurationContext, serviceAccount core.ServiceAccount, saCreds map[string]string, publicKey *keys.PublicKey) error {
-	for key, value := range saCreds {
-		path := "/sa-" + serviceAccount.Type + "/" + key
+func writeConfig(ctx context.Context, senDoguCfg *config.DoguConfig, cfgRepo SensitiveDoguConfigRepository) error {
+	update, err := cfgRepo.Update(ctx, *senDoguCfg)
```

Comment:
🚫 **[checkstyle]** <staticcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA4006: this value of `update` is never used
<!-- __reviewdog__:ChAyOWIyZmE3NzczMjkzOTRlEgpjaGVja3N0eWxl -->